mod/tagrm.php

```diff
@@ -52,23 +51,24 @@ function update_tags($item_id, $tags){
 		$a->internalRedirect($_SESSION['photo_return']);
 	}
 
-	$arr = explode(',', $item['tag']);
+	$old_tags = explode(',', $item['tag']);
 
-	foreach ($tags as $t) {
-		foreach ($arr as $i => $x) {
-			if (strcmp($x, $t) == 0) {
-				unset($arr[$i]);
+	foreach ($tags as $new_tag) {
+		foreach ($old_tags as $count => $old_tag) {
+			if (strcmp($old_tag, $new_tag) == 0) {
+				unset($old_tags[$count]);
 				break;
 			}
 		}
 	}
 
-	$tag_str = implode(',',$arr);
-	if(empty($tag_str)){
-		$tag_str = '';
+	$tag_str = implode(',',$old_tags);
+	if(!empty($tag_str)) {
+		Item::update(['tag' => $tag_str], ['id' => $item_id]);
```

Comment:
I just realized `update_tags($item_id, $tags)` should be removed and its calls replaced by `Term::insertFramTagFieldByItemId($item_id, $tags)` calls.

Reply:
Ah, you're right, but this module is confusing. Removing tags is the same as saving the remaining tags, so only the `Item::update` part and the conditional surrounding it.

Reply:
But wait. That means that `Term::insertFramTagFieldByItemId` needs to decide if it calls `Term::deleteByItemId($item_id);` or not, but that was the way I implemented it before.
Or do you mean something like:

	Term::insertFramTagFieldByItemId($item_id, $tag_str);
	if(!empty($tag_str)) {
		Term::deleteByItemId($item_id);
	}

But in that case `Term::insertFramTagFieldByItemId($item_id, $tag_str);` is even called when it isn't need to.

Reply:
Item tags are deleted in `Term::insertFromTagFieldByItemId` no matter what and it's fine.

Reply:
You are right! I didn't see that. Thanks :)